contracts/staking/src/contract.rs

```diff
@@ -412,23 +412,24 @@ mod tests {
     use std::str::FromStr;
 
     fn sample_validator(addr: &str) -> Validator {
-        Validator {
-            address: addr.to_owned(),
-            commission: Decimal::percent(3),
-            max_commission: Decimal::percent(10),
-            max_change_rate: Decimal::percent(1),
-        }
+        serde_json::from_value(serde_json::json!({
+            "address": addr.to_owned(),
+            "commission": Decimal::percent(3),
+            "max_commission": Decimal::percent(10),
+            "max_change_rate": Decimal::percent(1),
+        }))
+        .unwrap()
```

Comment:
Don't we have a constructor to do this in a safer way than deserialization?

Reply:
We have, but that could have new fields added at any time and I wanted to discourage contract authors from using it.
However, I see a problem with this now: Both `Validator` and `FullDelegation` are part of the [public `MockQuerier` api](https://github.com/CosmWasm/cosmwasm/blob/fc99133702607764e6da8f640ec9d27416e188e7/packages/std/src/testing/mock.rs#L525-L533), so we need a stable way to construct them. Maybe it would be better to add a stable constructor for them instead (and if anything is added, put Default values for that). WDYT?

Reply:
> WDYT?

🤯 we are walking in circles on that topic. This is requesting a new field to be added to Validator (https://github.com/CosmWasm/cosmwasm/issues/1828).

Reply:
Yes, but if I remember correctly #1828 is no longer needed for Mesh Security. Regardless of that, I don't find it inconceivable that the we would want to add fields to either of them (e.g. in the sdk, Validator has loads of fields).

However, some way to construct those two structs is needed because of that `MockQuerier::update_staking` function and I would prefer it being a stable way. Breaking testing frameworks that are deeply integrated with our code is one thing, but breaking normal users' test code seems bad to me. So, in this case I would just live with the fact that a newly added field might not have a sensible default value when constructed.

Reply:
Agreed. Then I guess we want two constructors, a stable one with the current fields and an unstable one (as generated by the macro) for all the fields. Right?

Reply:
Yes, that would probably be the most consistent way to do it